src/library/scala/collection/immutable/ListSet.scala

```diff
@@ -119,12 +119,8 @@ sealed class ListSet[A] extends AbstractSet[A]
     var that: ListSet[A] = self
     def hasNext = that.nonEmpty
     def next: A =
-      if (hasNext) {
-        val res = that.head
-        that = that.tail
-        res
-      }
-      else Iterator.empty.next()
+      if (!hasNext) throw new NoSuchElementException("next on empty iterator")
```

Comment:
You're right, I misunderstood what the current code was doing. Using `Iterator.empty.next()` is indeed a good way to do this!